src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.targets

```diff
@@ -215,18 +216,54 @@ Copyright (c) .NET Foundation. All rights reserved.
                                    Condition="'%(_NETCoreNativeItems.FileName)%(_NETCoreNativeItems.Extension)' == '$(_DotNetHostPolicyLibraryName)' or
                                               '%(_NETCoreNativeItems.FileName)%(_NETCoreNativeItems.Extension)' == '$(_DotNetHostFxrLibraryName)'" />
 
-      <NativeNETCoreCopyLocalItems Include="%(_NETCoreNativeItems.ResolvedPath)"
-                                   Condition="'%(_NETCoreNativeItems.FileName)%(_NETCoreNativeItems.Extension)' == '$(_DotNetHostExecutableName)'">
+      <NativeRestoredAppHostNETCore Include="%(_NETCoreNativeItems.ResolvedPath)"
+                                   Condition="'%(_NETCoreNativeItems.FileName)%(_NETCoreNativeItems.Extension)' == '$(_DotNetAppHostExecutableName)'"/>
+
+      <AllNETCoreCopyLocalItems Include="@(NativeNETCoreCopyLocalItems)">
+        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+        <CopyToPublishDirectory>Never</CopyToPublishDirectory>
+      </AllNETCoreCopyLocalItems>
+    </ItemGroup>
+
+    <PropertyGroup Condition="'@(NativeRestoredAppHostNETCore)' != '' ">
+      <AppHostDestinationDirectoryPath>$(BaseIntermediateOutputPath)\$(TargetFramework)\$(RuntimeIdentifier)\host</AppHostDestinationDirectoryPath>
+    </PropertyGroup>
+
+    <NETSdkError Condition="'@(NativeRestoredAppHostNETCore->Count())' &gt; 1"
+            ResourceName="MultipleFilesResolved"
+            FormatArguments="$(_DotNetAppHostExecutableName)" />
+    
+    <EmbedAppNameInHost   AppHostSourcePath="@(NativeRestoredAppHostNETCore)"
+                          AppHostDestinationDirectoryPath="$(AppHostDestinationDirectoryPath)"
+                          AppName="$(AssemblyName).dll"
```

Comment:
This is the extension of the managed assembly, i believe it will always be a ".dll"